app/helpers/sessions_helper.rb

```diff
@@ -4,15 +4,36 @@ def log_in user
   end
 
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id])
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by(id: user_id) if session[:user_id]
+    elsif user_id = cookies.signed[:user_id]
```

Comment:
signed có tác dụng gì

Reply:
Nếu đặt id dưới dạng văn bản thuần túy sẽ làm lộ dạng cookie của ứng dụng và khiến kẻ tấn công dễ dàng xâm nhập tài khoản người dùng. Để tránh sự cố này, ta sử dụng cookie signed, cookie này sẽ mã hóa cookie một cách an toàn trước khi đặt nó trên trình duyệt